events/_event-post-macros.html

```diff
@@ -104,23 +109,55 @@
 
    Create HTML event meta address markup when given:
 
-   post: A post date from a query result.
+   post                     : A post date from a query result.
+
+   address_format (optional): A python string format to display to 
+                              address information  
+                              (e.g. '{venue} - {city}, {state}').
 
    ========================================================================== #}
 
-{% macro event_meta_address(post) %}
-    {%- set city = (post.location or 'Washington DC') -%}
-    {%- set state = (post.state or '') -%}
-    {%- set venue = (post.venue or 'CFPB') -%}
+{% macro event_meta_address(post, address_format='{city} {state}') %}
+    {%- set city  =  post.city   | default('Washington', true) -%}
+    {%- set state =  post.state  | default('DC', true) -%}
+    {%- set street = post.street | default('1275 First St NE', true) -%}
+    {%- set venue  = post.venue  | default('CFPB', true) -%}
+    {%- set zip    = post.zip    | default('20002', true) -%}
+
     <p class="event-meta_address h-adr">
-        <span class="event-meta_city p-locality">{{ city }}</span>
-        <span class="event-meta_state p-region">{{ state }}</span>
-        <span class="event-meta_venue p-extended-address">{{ venue }}</span>
+        {%- macro _city() %}
+            <span class="event-meta_city p-locality">{{ city }}</span>
+        {% endmacro %}
+
+        {%- macro _state() %}
+            <span class="event-meta_state p-state">{{ state }}</span>
+        {% endmacro %}
+
+        {%- macro _street() %}
+            <span class="event-meta_street p-street-address">{{ street }}</span>
+        {% endmacro -%}
+
+        {% macro _venue() %}
+            <span class="event-meta_venue p-extended-address">{{ venue }}</span>
+        {% endmacro %}
+
+        {% macro _zip() %}
+            <span class="event-meta_zip p-postal-code">{{ zip }}</span>
+        {% endmacro %}
+        
+        {{
+            address_format.format(**{
+                'city'   : _city()   | trim,
+                'street' : _street() | trim,
+                'state'  : _state()  | trim,
```

Comment:
Using the default values this will be fed:

```
Washington
1275 First St NE
DC
CFPB
20002
```
Does `address_format.format` handle the order of those values correctly?